File: ObjectiveC/DFCI-Lib/ZoneChasseSecteur5.m
//
//  ZoneChasseSecteur5.m
//  DFCI-ObjC
//
//  Created by Patrice Trognon on 19/09/2015.
//  Copyright © 2015 Patrice Trognon. All rights reserved.
//

#import "ZoneChasseSecteur5.h"

#import "LambertII.h"
#import "Coord.h"

@implementation ZoneChasseSecteur5

-(instancetype)initWithNumeroZone:(int)numeroZone carre2Km:(Carre2Km*)carre2Km lambertOrigine:(LambertII*)lambert {
    self = [super init];
    if ( self ) {
        self.numeroZone = numeroZone ;
        self.carre2Km = carre2Km ;
        
        {
            LambertII *lambertIISO = [[LambertII alloc]initWithXLambert:lambert.x + 500
                                                               YLambert:lambert.y + 500];
            LatLng *latLngSO = [self latLngFromLambertII:lambertIISO];
            self.coordSO = [[Coord alloc]initWithLatLng:latLngSO lambertII:lambertIISO];
        }
        
        {
            LambertII *lambertIISE = [[LambertII alloc]initWithXLambert:lambert.x + 1500
                                                               YLambert:lambert.y + 500];
            LatLng *latLngSE = [self latLngFromLambertII:lambertIISE];
            self.coordSE = [[Coord alloc]initWithLatLng:latLngSE lambertII:lambertIISE];
        }
        
        {
            LambertII *lambertIINE = [[LambertII alloc]initWithXLambert:lambert.x + 1500
                                                               YLambert:lambert.y + 1500];
            LatLng *latLngNE = [self latLngFromLambertII:lambertIINE];
            self.coordNE = [[Coord alloc]initWithLatLng:latLngNE lambertII:lambertIINE];
        }
        
        {
            LambertII *lambertIINO = [[LambertII alloc]initWithXLambert:lambert.x + 500
                                                               YLambert:lambert.y + 1500 ];
            LatLng *latLngNO = [self latLngFromLambertII:lambertIINO];
            self.coordNO = [[Coord alloc]initWithLatLng:latLngNO lambertII:lambertIINO];
        }
        
        
        {
            LambertII *lambertIICentre = [[LambertII alloc]initWithXLambert:lambert.x + 1000
                                                                   YLambert:lambert.y + 1000];
            LatLng *latLngCentre = [self latLngFromLambertII:lambertIICentre];
            
            self.coordCentre = [[Coord alloc]initWithLatLng:latLngCentre lambertII:lambertIICentre];
        }
        

    }
    return self;
}

@end

File: Java/com/ptro/dfci/ZoneChasseSecteur5.java
package com.ptro.dfci;

public class ZoneChasseSecteur5 extends CarreDFCI {

	public		int			numeroZone;
	
	public		Coord		coordSO;
	public		Coord		coordNO;
	public		Coord		coordNE;
	public		Coord		coordSE;
	public		Coord		coordCentre;
	
	public		Carre2Km	carre2Km;

	public ZoneChasseSecteur5(int numeroZone, Carre2Km carre2Km, LambertII lambert) {
		{
			LambertII lambertIISO = new LambertII(lambert.getX() + 500, 
					  							  lambert.getY() + 500);
			LatLng latLngSO = latLngFromLambertII(lambertIISO);
			coordSO = new Coord(latLngSO, lambertIISO);
		}
		
		{
			LambertII lambertIISE = new LambertII(lambert.getX() + 1500, 
												  lambert.getY() + 500);
			LatLng latLngSE = latLngFromLambertII(lambertIISE);
			coordSE = new Coord(latLngSE, lambertIISE);
		}

		{
			LambertII lambertIINE = new LambertII(lambert.getX() + 1500, 
												  lambert.getY() + 1500);
			LatLng latLngNE = latLngFromLambertII(lambertIINE);
			coordNE = new Coord(latLngNE, lambertIINE);
		}

		{
			LambertII lambertIINO = new LambertII(lambert.getX() + 500, 
												  lambert.getY() + 1500);
			LatLng latLngNO = latLngFromLambertII(lambertIINO);
			coordNO = new Coord(latLngNO, lambertIINO);
		}

		{
			LambertII lambertIICentre = new LambertII(lambert.getX() + 1000, 
												  lambert.getY() + 1000);
			LatLng latLngCentre = latLngFromLambertII(lambertIICentre);
			coordCentre = new Coord(latLngCentre, lambertIICentre);
		}
	}

}
